corehq/apps/domain/models.py

```diff
@@ -559,6 +566,34 @@ def _get_by_name(stale=False):
         return domain
 
     @classmethod
+    @skippable_quickcache(['alias'], skip_arg='strict', timeout=30 * 60)
+    def get_by_alias(cls, alias, strict=False):
```

Comment:
I still think you can refactor this and `get_by_name` into a single private method and then just call it from here with the appropriate view name